include/rocksdb/cache.h

```diff
@@ -240,6 +244,15 @@ struct CompressedSecondaryCacheOptions : LRUCacheOptions {
   // into chunks so that they may better fit jemalloc bins.
   bool enable_custom_split_merge = false;
 
+  // Kinds of entries to allow into the cache when evicted from primary cache.
+  CacheEntryRoleSet include_entry_types = CacheEntryRoleSet::All();
```

Comment:
I guess there's no harm in having this option, but is there a scenario where you think this might be useful? If an object is cached in the primary cache, it means there's value in caching it. When would it make sense to exclude it from the secondary cache?

Reply:
I could foresee the vast majority of filter and index block evictions being because the file has become obsolete, meaning it's simply extra work trying to track it in the secondary cache. Also secondary cache might be involved in keeping metadata blocks on local storage, with a dedicated, segregated secondary cache for them.

However, this complete include vs. exclude could be implemented as a secondary cache wrapper class available in the public API. And then it could be applied to other secondary caches also. I think I'll split that off into another PR for consideration.

Reply:
> I could foresee the vast majority of filter and index block evictions being because the file has become obsolete, meaning it's simply extra work trying to track it in the secondary cache. Also secondary cache might be involved in keeping metadata blocks on local storage, with a dedicated, segregated secondary cache for them.
> 
> However, this complete include vs. exclude could be implemented as a secondary cache wrapper class available in the public API. And then it could be applied to other secondary caches also. I think I'll split that off into another PR for consideration.

Makes sense to have the complete include/exclude as a wrapper class, since its generally applicable. The `exclude_compression_entry_types` option is specific to the compressed secondary cache, so makes sense to have it here.